src/pkg/utils/general_test.go

```
@@ -32,9 +41,82 @@ func TestContainsMention(t *testing.T) {
+	meta := utils.GetMetaDataFromURL(server.URL)
+	assert.Equal(suite.T(), "Test Title", meta.Title)
+	assert.Equal(suite.T(), "Test Description", meta.Description)
+	assert.Equal(suite.T(), "http://example.com/image.jpg", meta.Image)
+}
+
+func (suite *UtilsTestSuite) TestDownloadImageFromURL() {
+	// Use httptest.NewServer to mock HTTP server
+	server := httptest.NewServer(http.HandlerFunc(func(w http.ResponseWriter, r *http.Request) {
+		if r.URL.Path == "/image.jpg" {
+			w.Header().Set("Content-Type", "image/jpeg") // Set content type to image
+			w.Write([]byte("image data"))
```

golangci-lint (1.62.2)

Error return value of `w.Write` is not checked

(errcheck)